### ISSUE TITLE
Allowing use of empty templates

### ISSUE THREAD

**Issue by IAmJulianAcosta** [miguelcobain/ember-cli-selectize#147]
Hi! I'm using selectize with a templatelike this:

```
{{#if model.isVisible}}
    {{model.foo}}
    ...
{{/if}}
``` 

But there is a problem when the model is not visible, if I not render something, `if (this._selectize.getOption(data.value).length !== 0) {` this condition will not met (https://github.com/miguelcobain/ember-cli-selectize/blob/master/addon/components/ember-selectize.js#L741) and everything will fail. 

Is there anyway to change this conditional?

**Comment by miguelcobain** [comment]
How/where are you using that template?

**Comment by IAmJulianAcosta** [comment]
```renderOption              : function (model) {
			let selectizeOptionComponent = SelectizeOption.create (
				{
					model : model,
					layout   : SelectizeOptionTemplate
				}
			);
			
			return ComponentToDom.getDOMElement (selectizeOptionComponent).innerHTML;
		},
```

**Comment by miguelcobain** [comment]
I honestly have no idea of what that code does, but it seems too complicated. You know that ember-cli-selectize supports custom components? Have you tried that? Also, why exactly do you need a template? What do you what to customize? What do you pretend to achieve? I need all this basic info or I won't be able to help you.

**Comment by IAmJulianAcosta** [comment]
> You know that ember-cli-selectize supports custom components? Have you tried that?

TBH, nope, didn't tried that, how can I do this?

**Comment by IAmJulianAcosta** [comment]
@miguelcobain Update:

I removed my custom template rendering (and added a quick fix to #152), in order to check if is the problem. Now I'm using `optionComponent` and `itemComponent` to render my component, but the problem continues. 

I have a list of models that has a property named `isVisible`. If `isVisible` is true, model should appear in selectize search, but if is not, it shouldn't appear, so I have a conditional in my component with the required markup. If `isVisible` is false, there isn't markup to render.

The problems come when model is not visible. I filter some models by some parameters and the list of visible elements in selectize now renders 3 models instead of 10. Now I filter by other parameter and selectize now renders 2 models. If I filter in again the model, `ember-selectize` will not include again in the list. 

This is caused by the code that I mentioned in the issue.
